dilated-file-transfer.md

```diff
@@ -0,0 +1,484 @@
+
+The `"mode"` key indicates the desired mode of the peer.
+It has one of three values:
+* `"send"`: the peer will only send files (similar to classic transfer protocol)
+* `"receive"`: the peer only receive files (the flip side of the above)
+* `"connect"`: the peer will send and receive zero or more files before closing the session
+
+Note that `"send"` and `"receive"` above will still use Dilation as all clients supporting this protocol must.
+If a peer sends no version information at all, it will be using the classic protocol (and is thus using Transit and not Dilation for the peer-to-peer connection).
+
+    XXX: maybe we don't strictly _need_ the mode at all? but it does keep things explicit .. which might be important for UX decisions on one or the other side?
+    XXX: that is:
+    XXX: a "receive-only" client can simply never send an offer
+    XXX: a "send-only" client simply (automatically) rejects any offer
+
+    XXX: "being explicit" has the advantage that a "receive-only" peer that connects to another "receive-only" peer will discover that quickly, and can (properly) fail -- otherwise, they'll only notice when one side gets bored and quits
+
+    XXX: two "send-only" peers that connect will fail fairly quickly -- each side should disconnect with a protocol-error when their receive the first offer
+
+    XXX: a "connect" peer that contacts either a "receive-only" or "send-only" can benefit from "being explicit" by disabling parts of the UI (for example) that won't work (e.g. contacting a "send-only" peer means the user can't browse/drop files)
+
+The `"features"` key points at a list of message-formats / features understood by the peer.
+This allows for existing messages to be extended, or for new message types to be added.
+Peers MUST _accept_ messages for any features they declare in `"features"`.
+Peers MUST only send messages / attributes for features in the other side's list.
+Only one format exists currently: `"core"`.
+
+   XXX:: maybe just lean on "version" for now? e.g. version `2` could introduce "features"?
+
+See "Example of Protocol Expansion" below for discussion about adding new attributes (including when we might increment the `"version"` instead of adding a new `"feature"`).
+
+The `"permission"` key specifies how to proceed with offers.
+Using mode `"ask"` tells the sender to pause after transmitting the first metadata message and await an answer from the peer before streaming further data.
+Using mode `"yes"` means the peer will accept all incoming transfers so the sender should not pause after the metadata (and instead immediately start sending data messages).
+Although a peer could implement "yes" mode by simply sending an accept message for each offer without user interaction, setting this mode cuts down latency for "one-way" transfers (see [Discussion](#discussion))
```

Comment:
I don't think latency is that critical here. Do you think saving that round trip is worth the added complexity?

Reply:
I don't have super strong feelings, this comes from some other "next file transfer protocol" discussions .. it could be relegated to "a future protocol update" too/instead.

Reply:
okay, I moved this to a "future enhancement" example (and it fills the "example the needs a new protocol version" slot too).